File: StrategyPattern/bank/MinusAccount.java
package bank;

public class MinusAccount extends Account
{
	private int bound;
	
	public MinusAccount(int bound)
	{
		super(new ConcreteWithdraw2(), new ConcreteCalculateInterest2());
		this.bound = bound;
	}
	
	public void setBound(int bound)
	{
		this.bound = bound;
	}
	
	public int getBound()
	{
		return bound;
	}
}

File: StrategyPattern_C++/MinusAccount.cpp
#include "MinusAccount.h"



MinusAccount::MinusAccount(int bound) : Account(new ConcreteWithdraw2(), new ConcreteCalculateInterest2())
{
	setBound(bound);
}

MinusAccount::~MinusAccount()
{
}

void MinusAccount::setBound(int bound)
{
	this->bound = bound;
}

int MinusAccount::getBound()
{
	return bound;
}
